src/fx/AnimateableTimeStretchPitchScaleReader.cpp

```diff
@@ -45,14 +45,45 @@ void AnimateableTimeStretchPitchScaleReader::read(int& length, bool& eos, sample
 
 void AnimateableTimeStretchPitchScaleReader::seek(int position)
 {
-	double time = double(position) / double(m_reader->getSpecs().rate);
+	double sampleRate = double(m_reader->getSpecs().rate);
+	double time = double(position) / sampleRate;
 	float frame = time * m_fps;
 
 	float timeRatio = m_timeStretch->readSingle(frame);
 	setTimeRatio(timeRatio);
 	float pitchScale = m_pitchScale->readSingle(frame);
 	setPitchScale(pitchScale);
-	TimeStretchPitchScaleReader::seek(position);
-}
 
+	int inputSamplePos = 0;
+	double outputSamplePos = 0.0;
+
+	float ratio = 1.0f;
+	float lastRatio = 1.0f;
+
+	const double frameDuration = 1.0 / m_fps;
+	const double samplesPerFrame = frameDuration * sampleRate;
+	int frameIndex = 0;
+
+	while(outputSamplePos < position)
+	{
+		ratio = m_timeStretch->readFrameSingle(frameIndex);
+
+		if(ratio <= 0.0f)
+			ratio = lastRatio;
+		else
+			lastRatio = ratio;
+
+		outputSamplePos += samplesPerFrame;
+		inputSamplePos += static_cast<int>(samplesPerFrame / ratio);
+
+		frameIndex++;
+	}
+
```

Comment:
As a final step, I'd call `readSingle` for the actual `position` and add `(position - total_frames * samplesPerFrame) / readSingle(...)` to `inputSamplePos` to account for the remainder.